### PR TITLE
Fix issue with e2e not starting [2790]

### PR DESCRIPTION
**Pull request by samnajian**
## Description
This PR  make sure `docker-compose` command exists before trying to setup the e2e environment, which relies on this.

### Related issue(s)

#2790 
### Steps to reproduce & screenshots/GIFs

If the e2e tests can run, the PR is green.

### Checklist

- [ ] unit tests

- [ ] `changelog.txt` entry added
- [ ] `readme.txt` entry added

### CODE REVIEW

File: .github/workflows/e2e_tests.yml [hunk 1]
```diff
@@ -43,6 +43,10 @@ jobs:
             composer install
             npm ci
             npm run dist
+      - name: Install docker-compose
+        run: |
+          sudo curl -L "https://github.com/docker/compose/releases/download/v2.3.3/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+          sudo chmod +x /usr/local/bin/docker-compose
```
Comment (Ferdev) [inline .github/workflows/e2e_tests.yml:49]:
```suggestion
          sudo chmod 550 /usr/local/bin/docker-compose
```

Comment (samnajian) [reply .github/workflows/e2e_tests.yml:49]: The change, although more secure, wouldn't change much in this context.

Comment (samnajian) [reply .github/workflows/e2e_tests.yml:49]: @Ferdev The change made the E2E setup fail, see https://github.com/Automattic/woocommerce-services/actions/runs/10366467965/job/28695765567, so I reverted it back.

Comment (Ferdev) [reply .github/workflows/e2e_tests.yml:49]: sounds good! Pretty pretty weird that GitHub is running `docker-composer` with a user distinct than the owner, or not even adding the user they're using to the owner's group

**Pull request merged**